sklearn/model_selection/_split.py

```diff
@@ -1769,6 +1769,36 @@ def __init__(self, *, n_splits=5, n_repeats=10, random_state=None):
             n_splits=n_splits,
         )
 
+    def split(self, X, y, groups=None):
+        """Generate indices to split data into training and test set.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Training data, where `n_samples` is the number of samples
+            and `n_features` is the number of features.
+
+        y : array-like of shape (n_samples,)
+            The target variable for supervised learning problems.
+
+        groups : object
+            Always ignored, exists for compatibility.
+
+        Yields
+        ------
+        train : ndarray
+            The training set indices for that split.
+
+        test : ndarray
+            The testing set indices for that split.
+        """
+        if groups is not None:
+            warnings.warn(
+                f"The groups parameter is ignored by {self.__class__.__name__}",
+                UserWarning,
+            )
```

Comment:
```suggestion
```

I think you can remove this since `super().split` will go to `_UnsupportedGroupCVMixin.split` which would do the warning about groups

Reply:
Also please add a comment in the method explaining that implemented this method is needed to make the `y` parameter required, since `_UnsupportedGroupCVMixin.split` has it optional or something like this

Reply:
Removed the lines from suggested change, yes it's check that in super class.

Also added a comment that y is required in the function definition.